weldx/util.py

```diff
@@ -23,6 +23,36 @@
     import weldx.transformations as tf
 
 
+def deprecated(func: Callable):
+    """Mark a functions as deprecated.
+
+    This decorator emits a warning when the function is used.
+
+    Notes
+    -----
+    Original source: https://stackoverflow.com/a/30253848/6700329
+
+    Parameters
+    ----------
+    func :
+        The wrapped function
+
+    """
+
+    @wraps(func)
+    def _new_func(*args, **kwargs):
+        warnings.simplefilter("always", DeprecationWarning)  # turn off filter
```

Comment:
If we go for a custom thing, we should derive from DeprecationWarning in order to not collide with other packages (which might raise a lot of these). Resetting the filter internally ain't a good thing IMO.

Reply:
It would also be great to have a simple message prepended which feature to use in the future.

Reply:
We can build more utility into the function. Just wanted to wait until we decided to go with our own approach before investing more time into it.

Reply:
I think we should go for a custom impl, since this feature demands not too much code and we should avoid adding more dependencies.

Reply:
Guess you are right. I'll add some useful features to the decorator.